src/datasources/system/SystemDataSource.ts

```diff
@@ -33,7 +34,11 @@ export class SystemDataSource extends DataSourceBase<SystemQuery> {
         ],
       };
     } else {
-      const metadata = await this.getSystemMetadata(this.templateSrv.replace(query.systemName, options.scopedVars));
+      const metadata = await this.getSystemMetadata(
+        this.templateSrv.replace(query.systemName, options.scopedVars),
+        defaultProjection,
+        this.templateSrv.replace(query.workspace, options.scopedVars)
+      );
```

Comment:
I'm getting an error if I put in both a system and a workspace in the panel query editor. The generated filter is:

`id = "asdf" || alias = "asdf" workspace = "e2897cf7-6332-433c-8284-65b7b628f3f6"`

That's my fault - I think the join in `getSystemMetadata` needs to use `&&` as the separator. Although now that I think about it, it's not much of a use case to filter by both a specific system and a workspace. I guess you could have systems with the same name in different workspaces?

Reply:
I agree that filtering by system and workspace isn't really needed. I'm pretty sure that minionIDs have to be unique at least at the salt level. The workspace filter is mainly needed when viewing the metadata for all systems as well as for the system variable to limit the systems in the drop-down.

Reply:
Okay, I'll look into only showing the workspace drop-down when "All systems" is selected.